src/Listeners/KernelEventsSubscriber.php

```diff
@@ -115,7 +114,22 @@ public function onKernelPostControllerArguments(ControllerArgumentsEvent $event)
 
         $this->endSegment(KernelEvents::CONTROLLER_ARGUMENTS);
 
-        $this->startSegment(self::SEGMENT_TYPE_PROCESS, self::SEGMENT_CONTROLLER);
+        $controllerLabel = $event->getRequest()->attributes->get('_controller');
+
+        $arguments = [];
+        foreach ($event->getArguments() as $argument) {
+            if (is_object($argument)) {
+                $args = ['class' => get_class($argument)];
+                if (method_exists($argument, 'getId')) {
+                    $args['id'] = $argument->getId();
+                }
+                $arguments[] = $args;
+            } else {
+                $arguments[] = $argument;
+            }
+        }
+        $segment = $this->startSegment(self::SEGMENT_TYPE_PROCESS, $controllerLabel);
+        $segment->addContext($controllerLabel, ['arguments' => $arguments]);
```

Comment:
We can simply use "Data" as context label.

Reply:
Better to use `arguments` as it is controller arguments